.github/workflows/code-quality.yml

```diff
@@ -0,0 +1,67 @@
+name: Analyse and format
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  analysis:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        php:
+          - 7.4
+          - 8.0
+          - 8.1
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite
+          tools: composer:v2,phpstan
+          coverage: none
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --no-interaction --no-suggest
```

Comment:
needed for phpstan, otherwise it can't resolve dependencies we use in the codebase